validphys2/src/validphys/n3fit_data.py

```diff
@@ -98,8 +101,10 @@ def tr_masks(data, replica_trvlseed):
         trmax = int(ndata * frac)
         if trmax == 0:
             # If that number is 0, then get 1 point with probability frac
-            trmax = int(rng.random() < frac)
-        mask = np.concatenate([np.ones(trmax, dtype=bool), np.zeros(ndata - trmax, dtype=bool)])
+            trmax = int(rng.random() < frac) if not parallel_models else 1
```

Comment:
This needs to either be fix or maybe add a warning for when running in parallel.

Reply:
You mean add a warning whenever we are running parallel replicas and have single-point datasets?

Reply:
Yes. Something like "putting {dstaset name} in the training set for all replicas" for the first replica.

But not a problem right now. You can leave these warnings for the end.

Reply:
Done in 522184dbe6bc12d7d8215eaefbc68cffdfe520ca. Also, when parallel models is set true but the no. replicas is one, the single-point datasets are again randomly distributed among training and validation sets.